Kuadrant/authorino-operator
# Support for Authorino enable/disable superseding strict host subsets

Exposes Authorino's `--allow-superseding-host-subsets` command-line flag (https://github.com/Kuadrant/authorino/pull/434) as a new API option `spec.supersedingHostSubsets: Boolean` (default: `false`)

### Verification steps

```sh
kind create cluster --name authorino-operator-test
kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.12.1/cert-manager.yaml
kubectl -n cert-manager wait --timeout=300s --for=condition=Available deployments --all
kubectl create namespace authorino-operator
make install
make run
```

In a separate shell:

```sh
kubectl apply -f -<<EOF
apiVersion: operator.authorino.kuadrant.io/v1beta1
kind: Authorino
metadata:
  name: authorino
spec:
  supersedingHostSubsets: true
  image: quay.io/kuadrant/authorino:host-collision-switch
  listener:
    tls:
      enabled: false
  oidcServer:
    tls:
      enabled: false
EOF
```

Confirm the `--allow-superseding-host-subsets` flag included in the command-line args of the Authorino container:

```sh
kubectl get deployment/authorino -o jsonpath='{.spec.template.spec.containers[0].args}'
# ["--watch-namespace=default","--allow-superseding-host-subsets"]%
```
